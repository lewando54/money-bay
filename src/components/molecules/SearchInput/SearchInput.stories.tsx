/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import SearchInput from './SearchInput'

const meta = {
    component: SearchInput,
    argTypes: {
        // To write
    }
} satisfies Meta<typeof SearchInput>

export default meta
type Story = StoryObj<typeof SearchInput>

export const Default: Story = {
    args: {
        // To write
    }
}
