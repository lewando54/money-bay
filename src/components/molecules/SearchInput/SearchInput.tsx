import React from 'react'
import { StyleProp, TextStyle,ViewStyle } from 'react-native'
import { Feather } from '@expo/vector-icons'

import { Input } from '@atoms'

import { globalColors } from '@styling/GlobalTheme.style'

interface ISearchInputProps {
    value?: string
    placeholder?: string
    containerStyle?: StyleProp<ViewStyle>
    textStyle?: StyleProp<TextStyle>
    onChange?: (newValue: string) => void
    onIconPress?: () => void
    testID?: string
}

export default function SearchInput({
    value,
    placeholder='Search',
    containerStyle,
    textStyle,
    onChange,
    onIconPress,
    testID
}: ISearchInputProps){
    return (
        <Input 
            value={value}
            placeholder={placeholder}
            containerStyle={containerStyle}
            textStyle={textStyle}
            icon={<Feather name="search" size={16} color={globalColors.MAIN_DARK} />}
            isIconVisible={true}
            onChange={onChange}
            onIconPress={onIconPress}
            testID={testID}
        />
    )
}
