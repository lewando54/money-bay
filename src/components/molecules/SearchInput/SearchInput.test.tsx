import React from 'react'
import renderer from 'react-test-renderer'

import SearchInput from './SearchInput'
describe('SearchInput', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<SearchInput />)
        expect(tree).toMatchSnapshot()
    })
})
