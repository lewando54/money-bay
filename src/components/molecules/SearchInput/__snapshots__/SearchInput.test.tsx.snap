// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#161966",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      [
        {
          "alignItems": "center",
          "borderColor": "white",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 14,
        },
        undefined,
        false,
      ],
    ]
  }
  testID="universal-container"
>
  <TextInput
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    placeholderTextColor="#868698"
    secureTextEntry={false}
    style={
      [
        {
          "color": "white",
          "fontFamily": "MulishRegular",
        },
        {
          "color": "#1B1D4D",
          "marginRight": 10,
          "maxWidth": "93%",
          "width": "100%",
        },
        undefined,
        false,
      ]
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={10}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={false}
    testID="input-icon"
  >
    <Text />
  </View>
</View>
`;
